library/public/utils.h

```diff
@@ -57,6 +59,28 @@ class F3D_EXPORT utils
   {
     explicit tokenize_exception(const std::string& what = "");
   };
+
+  /** String template allowing substitution of variables enclosed in curly braces.
+    ```
+    StringTemplate("{greeting} {name}!")
+      .substitute({ { "greeting", "hello" }, { "name", "World" } })
+      .str() == "hello World!"
+    ```
+   */
+  class StringTemplate
```

Comment:
maybe you need a F3D_EXPORT here ?

Reply:
thanks :)

Windows CI now has a compilation error instead of a link error https://github.com/f3d-app/f3d/actions/runs/12134916053/job/33833013791?pr=1744#step:3:1074
I assume that's progress but I have no idea what to do about it :/